### ISSUE TITLE
[fxa-db-server] exit with a non-zero code for "Error: Handshake inactivity timeout"

### ISSUE THREAD

**Issue by vladikoff** [mozilla/fxa#1386]
We've spotted a case where a `"Error: Handshake inactivity timeout"` timeout may cause the db-server to exit with an exit code 0. We should fix it so that exit code is non-zero.

cc @jbuck 

Some stack:

> at Timer.processTimers (timers.js:223:10)"
"    --------------------"
"    at Protocol._enqueue (/app/node_modules/mysql/lib/protocol/Protocol.js:144:48)"
"    at Protocol.handshake (/app/node_modules/mysql/lib/protocol/Protocol.js:51:23)"
"    at PoolConnection.connect (/app/node_modules/mysql/lib/Connection.js:118:18)"
"    at Pool.getConnection (/app/node_modules/mysql/lib/Pool.js:48:16)"
"    at PoolCluster._getConnection (/app/node_modules/mysql/lib/PoolCluster.js:229:13)"
"    at PoolNamespace.getConnection (/app/node_modules/mysql/lib/PoolNamespace.js:40:11)"
"    at PoolCluster.getConnection (/app/node_modules/mysql/lib/PoolCluster.js:145:13)"
"    at PoolCluster.tryCatcher (/app/node_modules/bluebird/js/release/util.js:16:23)"
"    at MySql.ret [as getClusterConnection] (eval at makeNodePromisifiedEval (/app/node_modules/bluebird/js/release/promisify.js:184:12), <anonymous>:14:23)"
"    at retryable (/app/lib/db/mysql.js:1227:12)"
"    at P (/app/lib/db/mysql.js:1158:7)"
"    at Promise._execute (/app/node_modules/bluebird/js/release/debuggability.js:313:9)"
"    at Promise._resolveFromExecutor (/app/node_modules/bluebird/js/release/promise.js:483:18)"
"    at new Promise (/app/node_modules/bluebird/js/release/promise.js:79:10)"
"    at MySql.getConnection (/app/lib/db/mysql.js:1157:12)"
"    at MySql.singleQuery (/app/lib/db/mysql.js:961:17)"
"    at MySql.read (/app/lib/db/mysql.js:1085:17)"
"    at MySql.readFirstResult (/app/lib/db/mysql.js:1066:17)"
"    at P.resolve.then (/app/lib/db/mysql.js:159:20)"
"    at tryCatcher (/app/node_modules/bluebird/js/release/util.js:16:23)"
"    at Promise._settlePromiseFromHandler (/app/node_modules/bluebird/js/release/promise.js:512:31)"

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/FXA-118)
┆Issue Number: FXA-118

**Comment by clouserw** [comment]
This is two parts:

- We should fix the configuration to restart_always
- We should check the error code and fail

**Comment by jrgm** [comment]
I added the systemd change to Restart=always in https://github.com/mozilla-services/cloudops-deployment/pull/3231. 

I looked at the issue here and how to fix here. There error is emitted on a timeout deep down that I guess needs to be listened for in lib/db/mysql. Simple STR: `MYSQL_HOST=nosuch.host.name MYSQL_PASSWORD='' MYSQL_PORT=3306 node ./bin/server.js`. (`nosuch.host.name` could also be a real domain name, e.g., accounts.stage.mozaws.net. It just has to not be something that responds on port 3306).
